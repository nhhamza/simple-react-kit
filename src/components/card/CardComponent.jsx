import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import LibraryAddOutlinedIcon from '@material-ui/icons/LibraryAddOutlined';
import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import styles from './style.jsx';

const CardComponent = ({
  isCategory,
  title,
  subTitle,
  description,
  alt,
  height,
  image,
}) => {
  const classes = makeStyles(styles)();

  return (
    <Card className={classes.card}>
      {!isCategory && (
        <div className={classes.cardIconGroup}>
          <IconButton
            color="inherit"
            aria-label="menu"
            className={classes.cardIconPadding}
          >
            <FileCopyOutlinedIcon className={classes.cardIcon} />
          </IconButton>

          <IconButton
            color="inherit"
            aria-label="menu"
            className={classes.cardIconPadding}
          >
            <LibraryAddOutlinedIcon className={classes.cardIcon} />
          </IconButton>
        </div>
      )}
      <CardActionArea>
        <CardMedia alt={alt} height={height} image={image} component="img" />
        <CardContent>
          {isCategory ? (
            <Typography gutterBottom className={classes.cardTitleCategory}>
              {title}
            </Typography>
          ) : (
            <>
              <Typography gutterBottom className={classes.cardTitle}>
                {title}
              </Typography>
              <Typography gutterBottom className={classes.subTitle}>
                {subTitle}
              </Typography>
              <Typography gutterBottom className={classes.description}>
                {description}
              </Typography>
            </>
          )}
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

CardComponent.propTypes = {
  isCategory: PropTypes.bool,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  description: PropTypes.string,
  alt: PropTypes.string,
  height: PropTypes.string,
  image: PropTypes.string,
};

export default CardComponent;
