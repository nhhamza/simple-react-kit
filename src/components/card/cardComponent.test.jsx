import React from 'react';
import { mount } from 'enzyme';
import Typography from '@material-ui/core/Typography';
import imageTest from '../../assets/images/image1.png';
import CardComponent from './CardComponent.jsx';

describe('<CardComponent />', () => {
  const MOCK_PROPS = {
    isCategory: false,
    title: 'Perspective exterieure',
    subTitle: '00019 - Emile Ripert',
    description: 'Lille',
    alt: 'Contemplative Reptile',
    height: '140',
    image: imageTest,
  };

  const wrapper = mount(<CardComponent {...MOCK_PROPS} />);

  it('should render card component', () => {
    expect(wrapper.exists()).toMatchSnapshot();
  });

  it('should render card with title when isCategory is true', () => {
    const MOCK_PROP = {
      isCategory: true,
      title: 'Perspective exterieure',
      subTitle: '00019 - Emile Ripert',
      description: 'Lille',
      alt: 'Contemplative Reptile',
      height: '140',
      image: imageTest,
    };
    const newWrapper = mount(<CardComponent {...MOCK_PROP} />);
    expect(
      newWrapper
        .find(Typography)
        .first()
        .text(),
    ).toEqual('Perspective exterieure');
  });
});
