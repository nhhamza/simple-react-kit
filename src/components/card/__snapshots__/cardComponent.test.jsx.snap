// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardComponent /> should render card component 1`] = `true`;
